duplicate_enum_cases and conditional compilation

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The newly added `duplicate_enum_cases` does not consider conditional compilation flags.
It reports duplicates even though there are none in the configuration actually compiled.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint duplicate_enum_cases_if_else.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths duplicate_enum_cases_if_else.swift
Linting 'duplicate_enum_cases_if_else.swift' (1/1)
duplicate_enum_cases_if_else.swift:3:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
duplicate_enum_cases_if_else.swift:5:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
duplicate_enum_cases_if_else.swift:7:6: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
Done linting! Found 3 violations, 3 serious in 1 file.
```

### Environment

0.33.0
Cocoapods

```swift
enum Tag: String {
#if CONFIG_A
case value = "CONFIG_A"
#elseif CONFIG_B
case value = "CONFIG_B"
#else
case value = "CONFIG_DEFAULT"
#endif
}
```

--- a/Source/SwiftLintFramework/Rules/Lint/DuplicateEnumCasesRule.swift
+++ b/Source/SwiftLintFramework/Rules/Lint/DuplicateEnumCasesRule.swift
@@ -24,6 +24,17 @@ public struct DuplicateEnumCasesRule: ConfigurationProviderRule, SwiftSyntaxRule
             enum B {
                 case add(image: UIImage)
             }
+            """),
+            Example("""
+            enum Tag: String {
+            #if CONFIG_A
+                case value = "CONFIG_A"
+            #elseif CONFIG_B
+                case value = "CONFIG_B"
+            #else
+                case value = "CONFIG_DEFAULT"
+            #endif
+            }
             """)
         ],
         triggeringExamples: [

